compiler/rustc_mir_dataflow/src/impls/storage_liveness.rs

```diff
@@ -283,15 +282,15 @@ impl<'tcx> Analysis<'tcx> for MaybeRequiresStorage<'_, 'tcx> {
 
 impl<'tcx> MaybeRequiresStorage<'_, 'tcx> {
     /// Kill locals that are fully moved and have not been borrowed.
-    fn check_for_move(&mut self, trans: &mut <Self as Analysis<'tcx>>::Domain, loc: Location) {
-        let body = self.borrowed_locals.body();
-        let mut visitor = MoveVisitor { trans, borrowed_locals: &mut self.borrowed_locals };
-        visitor.visit_location(body, loc);
+    fn check_for_move(&self, trans: &mut <Self as Analysis<'tcx>>::Domain, loc: Location) {
+        let borrowed_locals = self.borrowed_locals.clone().into_results_cursor(self.body);
+        let mut visitor = MoveVisitor { trans, borrowed_locals };
+        visitor.visit_location(self.body, loc);
```

Comment:
This changes the complexity of the analysis. The former code use a cursor, which remembers the current position. The latter code recreates a cursor, which implicitly needs to recompute results from bb0. Could we use a `RefCell` or something?

Reply:
Hmm, ok. I've reinstated the cursor within a `RefCell`.